dbfit-java/build.gradle

```diff
@@ -303,65 +430,39 @@ project('netezza') {
         runtime 'org.netezza:nzjdbc3:7.2@jar'
     }
 
-    ext.connectionPropertiesFile = 'TestDbConnectionDbFitNetezza.properties'
-    ext.connectionPropertiesFiles = [connectionPropertiesFile]
+    ext.connectionPropertiesFiles = ['TestDbConnectionDbFitNetezza.properties']
+    ext.adminConnectionPropertiesFile = 'TestDbConnectionNetezzaAdmin.properties'
 
-    task connectionProperties {
-        doLast {
-            copyConnectionProperties '.'
-            def props = new Properties()
-            file(connectionPropertiesFile).withInputStream { props.load(it) }
-            def propService = props.getProperty('service')
-            def propServiceParts = propService.tokenize(':')
-            def propServiceHost = propServiceParts[0]
-            def propServicePort = propServiceParts[1]
-            def propDatabase = props.getProperty('database')
-
-            ext {
-                jdbcDriverClass = 'org.netezza.Driver'
-                jdbcUrlPrefix = 'jdbc:netezza://'
-                delimiterType = 'row'
-                userName = props.getProperty('username')
-                passWord = props.getProperty('password')
-                database = propDatabase.isEmpty() ? "" : ('/' + propDatabase)
-                jdbcUrl = jdbcUrlPrefix + propServiceHost + (propServicePort.isEmpty() ? (':' + propServicePort) : '')
-            }
+    deploymentProperties.doLast {
+        ext {
+            jdbcDriverClass = 'org.netezza.Driver'
+            jdbcDriverUrlPrefix = 'jdbc:netezza://'
+            delimiterType = 'row'
+            autoCommit = true
         }
     }
 
-    integrationTest.dependsOn connectionProperties
+    dbConnectionProperties.dependsOn(connectionFileProperties)
 
-    integrationTest.doFirst {
-        dbdeploy(connectionProperties.jdbcDriverClass,
-            connectionProperties.jdbcUrl + connectionProperties.database,
-            connectionProperties.userName,
-            connectionProperties.passWord,
-            connectionProperties.delimiterType)
+    def getJdbcUrl = {
+        service, database ->
+            return deploymentProperties.jdbcDriverUrlPrefix + service + '/' + database
     }
 
-    ext.execSqlScriptTestDatabase = {
-        scriptFile ->
-        sql(connectionProperties.jdbcDriverClass,
-            connectionProperties.jdbcUrl + '/system',
-            connectionProperties.userName,
-            connectionProperties.passWord,
-            scriptFile,
-            connectionProperties.delimiterType, ';', true)
-    }
-
-    task setUpTestDatabase(dependsOn: connectionProperties) {
-        doLast {
-            execSqlScriptTestDatabase(
-                "${rootProject.projectDir}/test_vm/scripts/netezza/sql/create-databases-netezza.sql")
+// can we add these defs to all subprojects (and also remove from TD and NZ)?
```

Comment:
What options do we have? So far I can think of:
* What you suggested - to have these properties always defined (with default of null or something)
* `ext.jdbcUrl = { getJdbcUrl(...) }`. That would need a pair of parentheses to get evaluated `jdbcUrl()`. I might be getting the syntax wrong but the general idea is to have that evaluated lazily.

I'm also wondering about defining `getJdbcUrl` as closure vs something with standard parameter list like `def getJdbcUrl(service, database)`. I suspect it would still fail to resolve the dependent properties (maybe even earlier), just I got curious what's the difference since I'm not sure.